### ISSUE TITLE
Upgrade: could not change blocking status for stream: Bad file descriptor

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#418]
When I try to upgrade sozu multiple times, the master process will show this error multiple times too:
```
2018-03-27T11:32:12Z 118284246427074 3025 MASTER DEBUG  sozu::command::orders   listener is: 9
2018-03-27T11:32:12Z 118284246461810 3025 MASTER DEBUG  sozu::command::orders   registering: Err(Error { repr: Os { code: 9, message: "Bad file descriptor" } })                                                 
2018-03-27T11:32:12Z 118284248249340 3025 MASTER ERROR  could not change blocking status for stream: Error { repr: Os { code: 9, message: "Bad file descriptor" } }                                       
2018-03-27T11:32:12Z 118284248266688 3025 MASTER DEBUG  sozu::command::orders   registering: Ok(())                                                                                                              
2018-03-27T11:32:12Z 118284249940237 3025 MASTER ERROR  could not change blocking status for stream: Error { repr: Os { code: 9, message: "Bad file descriptor" } }                                              
2018-03-27T11:32:12Z 118284249954589 3025 MASTER DEBUG  sozu::command::orders   registering: Err(Error { repr: Os { code: 17, message: "File exists" } })                                                
2018-03-27T11:32:12Z 118284251663292 3025 MASTER ERROR  could not change blocking status for stream: Error { repr: Os { code: 9, message: "Bad file descriptor" } }       
2018-03-27T11:32:12Z 118284251784419 3025 MASTER INFO   starting new master loop   
```

And sozuctl will print old workers still in the stopping state (I think that's related):
```
~ $ ../target/debug/sozuctl -c ../bin/my-config.toml upgrade
Preparing to upgrade proxy...

+--------+------+-----------+
| Worker | pid  | run state |
+--------+------+-----------+
| 1      | 2634 | Stopping  |
+--------+------+-----------+
| 2      | 2863 | Stopping  |
+--------+------+-----------+
| 0      | 2389 | Stopping  |
+--------+------+-----------+
| 3      | 3028 | Running   |
+--------+------+-----------+
```
At some point, the upgrade will hang. I only have one worker started each time.

To reproduce, I just upgrade a few times, waiting for sozu to be ready between each upgrade.

**Comment by Geal** [comment]
hello, does this still appear?

**Comment by BlackYoup** [comment]
Yes

**Comment by Geal** [comment]
this should be fixed by 6d935c4

**Comment by BlackYoup** [comment]
The message doesn't appear anymore :+1:
